src/main/resources/schema/provider.definition.schema.v1.json

```diff
@@ -210,6 +210,11 @@
         }
     },
     "type": "object",
+    "patternProperties": {
+        "^\\$id$": {
+            "$ref": "https://json-schema.org/draft-07/schema#/properties/$id"
+        }
+    },
```

Comment:
why is this expressed via patternProperties? Couldn't there just be a `"$id"` key under properties?

Reply:
we tried that and it doesn't work, because `$id` is a valid field that can appear anywhere: https://github.com/aws-cloudformation/cloudformation-cli/pull/344#issuecomment-554839461